content/blog/set-up-apache-skywalking-for-k8s-monitoring-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,368 @@
+
+## Prerequisites
+
+Before I start, make sure the following requirements are met:
+
+* A Kubernetes cluster, being provisioned in HPE GreenLake for Private Cloud Enterprise;
+* The *kubectl*, together with the *HPE kubectl plugin* and the *kubeconfig* file of the Kubernetes cluster. You can download them from the launched service console in HPE GreenLake for Private Cloud Enterprise. To simplify the setup process, you can export the environment variable `KUBECONFIG` and point it to the downloaded *kubeconfig* file;
+* The *[Helm](https://helm.sh/docs/intro/install/)* CLI tool, version 3.8.1 or later. 
+
+With your user access setup, you should have access to permissions that can create and update the following resources in the Kubernetes cluster:
+
+* Kubernetes Service Account(s)
+* Kubernetes Roles & RoleBindings
+
+## Set up Apache SkyWalking for Kubernetes monitoring
+
+Apache SkyWalking leverages Kubernetes [kube-state-metrics](https://github.com/kubernetes/kube-state-metrics) service for collecting metrics data from Kubernetes cluster. It then leverages *OpenTelemetry* collector to transfer the Kubernetes metrics to *OpenTelemetry* receiver in Apache SkyWalking Observability Analysis Platform (OAP) for Kubernetes monitoring. 
+
+![](/img/otel-collector.png)
+
+### Deploy Apache SkyWalking
+
+I will take the approach to setting up the Apache SkyWalking as a *self-managed* APM tool within the Kubernetes cluster created in HPE GreenLake for Private Cloud Enterprise. This mainly takes into account the security concerns in HPE GreenLake product environment. 
+
+Install Apache SkyWalking using Helm charts with *elasticsearch* as storage:
+
+```markdown
+$ git clone https://github.com/apache/skywalking-kubernetes 
+$ cd skywalking-kubernetes/chart
+$ helm repo add elastic https://helm.elastic.co
+$ helm dep up skywalking
+$﻿ kubectl create ns skywalking
+$ helm install skywalking skywalking –n skywalking \
+--set oap.image.tag=9.2.0 \
+--set oap.storageType=elasticsearch \
+--set ui.image.tag=9.2.0 \
+--set elasticsearch.imageTag=7.1.1 \7
+--set elasticsearch.persistence.enabled=true \
+--set elasticsearch.sysctlInitContainer.enabled=false \
+--set oap.env.SW_OTEL_RECEIVER=default \
+--set oap.env.SW_OTEL_RECEIVER_ENABLED_OC_RULES="k8s-cluster\,k8s-service\,k8s-node" 
+```
+
+After running above commands, t﻿he Apache SkyWalking is installed to the Kubernetes cluster's namespace *skywalking*. It creates the *elasticsearch* as the `StatefulSet` resource, running a pod on each worker node. It runs the Apache SkyWalking OAP with replicas as 2 to provide high availability.
+
+T﻿he last two options, *oap.env.SW_OTEL_RECEIVER=default* & *oap.env.SW_OTEL_RECEIVER_ENABLED_OC_RULES="k8s-cluster,k8s-service,k8s-node"*, enable the *OpenTelemetry* receiver and define the metrics for Kubernetes service, service instance and endpoint. It requires Apache SkyWalking OAP to have access to Kubernetes API server to query the metadata.
```

Comment:
Add the word "the" prior to "Kubernetes service" so it reads "...metrics for the Kubernetes service, service instance and endpoint."

Add "the" prior to "Kubernetes API server to query the metadata."